spec/lib/vmdb/config_spec.rb

```diff
@@ -2,14 +2,24 @@
   let(:password) { "pa$$w0rd" }
   let(:enc_pass) { MiqPassword.encrypt(password) }
 
-  it ".get_file" do
+  it ".get_file for current server" do
+    _guid, miq_server, zone = EvmSpecHelper.local_guid_miq_server_zone
     stub_settings(:http_proxy => {:default => {:host => "proxy.example.com", :user => "user", :password => password, :port => 80}})
 
     expect(VMDB::Config.get_file).to eq(
       "---\n:http_proxy:\n  :default:\n    :host: proxy.example.com\n    :user: user\n    :password: #{enc_pass}\n    :port: 80\n"
     )
   end
 
+  it ".get_file for sepecified resource" do
+    resource = FactoryGirl.create(:miq_server)
+    stub_settings(:http_proxy => {:default => {:host => "proxy.example.com", :user => "user", :password => password, :port => 80}})
```

Comment:
stub_settings should probably not be used here because the way it works is by stubbing things globally.

Instead, I think it's better to maybe create some changes, save them, then see if they come back?   This should be done in both tests.

Maybe something like...

```ruby
changes = {:http_proxy => {:default => {:host => "proxy.example.com", :user => "user", :password => password, :port => 80}}}

resource = FactoryGirl.create(:miq_server)
Vmdb::Settings.save!(miq_server, changes)

yaml = VMDB::Config.get_file(resource)

expect(YAML.load(yaml)).to include(changes)
```

Reply:
@h-kataria I edited the snippet above to be more proper